src/main/java/de/tum/in/www1/artemis/web/rest/LearningPathResource.java

```diff
@@ -399,11 +405,56 @@ public ResponseEntity<List<LearningPathNavigationObjectDTO>> getLearningObjectsF
 
         checkLearningPathAccessElseThrow(Optional.of(learningPath.getCourse()), learningPath, Optional.of(user));
 
-        List<LearningPathNavigationObjectDTO> learningObjects = learningPathRecommendationService.getOrderOfLearningObjectsForCompetency(competencyId, user).stream()
-                .map(learningObject -> LearningPathNavigationObjectDTO.of(learningObject, learningObjectService.isCompletedByUser(learningObject, user), competencyId)).toList();
+        List<LearningPathNavigationObjectDTO> learningObjects = learningPathRecommendationService.getOrderOfLearningObjectsForCompetency(competencyId, user, learningPathId)
+                .stream().map(learningObject -> LearningPathNavigationObjectDTO.of(learningObject, learningObjectService.isCompletedByUser(learningObject, user), competencyId))
+                .toList();
         return ResponseEntity.ok(learningObjects);
     }
 
+    /**
+     * GET courses/:courseId/learning-paths/configuration : Gets the learning paths configuration for the course
+     *
+     * @param courseId the id of the course for which to get the learning paths configuration
+     * @return the ResponseEntity with status 200 (OK) and with the configuration in the body
+     */
+    @GetMapping("courses/{courseId}/learning-paths/configuration")
+    @FeatureToggle(Feature.LearningPaths)
+    @EnforceAtLeastInstructorInCourse
+    public ResponseEntity<LearningPathsConfiguration> getLearningPathsConfiguration(@PathVariable long courseId) {
```

Comment:
It's probably not super important given the complexity of the LearningPathsConfiguration, but could we use a DTO instead? As I see it, we currently provide the `Course` along with the configuration to the client